homeassistant/components/lastfm/sensor.py

```diff
@@ -54,15 +55,22 @@ class LastfmSensor(Entity):
 
     def __init__(self, user, lastfm_api):
         """Initialize the sensor."""
+        self._unique_id = uuid.uuid4()
```

Comment:
I'm guessing this is wrong right? and should use something deterministic based on the user right? 

Any suggestions? Are hash functions usually used here?

Reply:
Since each entity represents one user, I'd use the username as unique_id. If the user changes username it means another account ie a new entity.

The unique_id must be unique and stable between restarts and the user should not be able to change unique_id for the same device or service.